ompi/mca/mtl/ofi/mtl_ofi_component.c

```diff
@@ -107,26 +135,70 @@ ompi_mtl_ofi_progress_no_inline(void)
 	return ompi_mtl_ofi_progress();
 }
 
+static int
+is_in_list(char **list, char *item)
+{
+    int i = 0;
+
+    if ((NULL == list) || (NULL == item)) {
+        return 0;
+    }
+
+    while (NULL != list[i]) {
+        if (0 == strncmp(item, list[i], strlen(item))) {
+            return 1;
+        } else {
+            i++;
+        }
+    }
+
+    return 0;
+}
+
 static struct fi_info*
 select_ofi_provider(struct fi_info *providers)
 {
+    char **include_list = NULL;
+    char **exclude_list = NULL;
     struct fi_info *prov = providers;
-    /**
-     * Unless explicitly asked by user, ignore the following OFI providers:
-     *     - sockets
-     *     - mxm
-     */
-    if (NULL == ompi_mtl_ofi.provider_name) {
-        while (NULL != prov) {
-            if ((0 == strncmp(prov->fabric_attr->prov_name, "sockets", 7)) ||
-                (0 == strncmp(prov->fabric_attr->prov_name, "mxm", 3))) {
-                prov = prov->next;
-                continue;
-            } else {
-                break;
-            }
+
+    opal_output_verbose(1, ompi_mtl_base_framework.framework_output,
+                        "%s:%d: mtl:ofi:provider_include = \"%s\"\n",
+                        __FILE__, __LINE__, prov_include);
+    opal_output_verbose(1, ompi_mtl_base_framework.framework_output,
+                        "%s:%d: mtl:ofi:provider_exclude = \"%s\"\n",
+                        __FILE__, __LINE__, prov_exclude);
+
+    if (NULL != prov_include) {
+        include_list = opal_argv_split(prov_include, ',');
+        while ((NULL != prov) &&
+               (!is_in_list(include_list, prov->fabric_attr->prov_name))) {
+            opal_output_verbose(1, ompi_mtl_base_framework.framework_output,
+                                "%s:%d: mtl:ofi: \"%s\" not in include list\n",
+                                __FILE__, __LINE__,
+                                prov->fabric_attr->prov_name);
+            prov = prov->next;
+        }
+    } else if (NULL != prov_exclude) {
+        exclude_list = opal_argv_split(prov_exclude, ',');
+        while ((NULL != prov) &&
+               (is_in_list(exclude_list, prov->fabric_attr->prov_name))) {
+            opal_output_verbose(1, ompi_mtl_base_framework.framework_output,
+                                "%s:%d: mtl:ofi: \"%s\" in exclude list\n",
+                                __FILE__, __LINE__,
+                                prov->fabric_attr->prov_name);
+            prov = prov->next;
```

Comment:
Very good, I'll update it.

(I didn't know about the comma. Thank you for pointing it out. -- As you can guess, English is not my native language :-P)